docs/book/src/migration/guide.md

```diff
@@ -0,0 +1,192 @@
+# Migration Guide
+
+This document describes how to migrate a project from kubebuilder v1 to v2.
+
+We suggest you read [this doc](./v1vsv2.md) that compares the notable differences between v1 and v2.
+
+The recommended way of migration is to create a new v2 project and copy the business logic code.
+
+Please ensure you have followed the [installation guide](../quick-start.md#installation) to install the required
+components.
+
+We have an [example v1 project](https://github.com/kubernetes-sigs/kubebuilder/tree/master/docs/book/src/migration/testdata/gopath/project). It should look like the
+[v2 project in the cronjob tutorial](https://github.com/kubernetes-sigs/kubebuilder/tree/master/docs/book/src/cronjob-tutorial/testdata/project) after the migration.
+
+## Preparation
+
+First, we need to figure out what are the domain, group, version and kind for kubebuilder.
```

Comment:
Lang: this reads awkwardly.  Generally, we say `what the XYZ are` when not asking something:

> What are the group, version, and domain?
> We'll need to figure out what the group, version, and domain are.